### ISSUE TITLE
Add note that "allowed_domains" should be a list of domains, not URLs

### ISSUE THREAD

**Issue by redapple** [scrapy/scrapy#2250]
(just logging the issue before I forget)

It may seem obvious by the name of the attribute that `allowed_domains` is about domain names, but it's not uncommon for scrapy users to make the mistake of doing `allowed_domains = ['http://www.example.com']`

I believe it is worth adding a note in http://doc.scrapy.org/en/latest/topics/spiders.html?#scrapy.spiders.Spider.allowed_domains

**Comment by eliasdorneles** [comment]
What if instead of simply documenting, Scrapy detect this case and issues a warning?

Even better, it could extract the domain from the URL and use that, while issuing a warning like:

    logging.warn("WARNING: allowed_domains accepts only domains, not URLs. Using allowed_domains = %r" % effective_allowed_domains)

**Comment by redapple** [comment]
+1 to issue a warning.

I'm less sure about inferring domain, for example for `http://www.example.com`, should it infer `example.com` or `www.example.com`?

**Comment by eliasdorneles** [comment]
I think the user expectation would be `www.example.com`.

I think it would be reasonable to use whatever urlparse parses as netloc:

```
>>> import urlparse
>>> urlparse.urlparse('http://www.example.com')
ParseResult(scheme='http', netloc='www.example.com', path='', params='', query='', fragment='')
```

**Comment by redapple** [comment]
Are you ok with having the new behavior as a seperate issue?

**Comment by eliasdorneles** [comment]
Sure, inferring a domain from an URL in allowed_domains looks like a separate feature indeed.

But I think warning users about URLs detected in allowed_domains and warning users about it belongs to the same issue as documenting (this one).
I mean, it isn't great to point users to docs when we can give them instant feedback.

Makes sense?

**Comment by redapple** [comment]
It does

**Comment by jlong49** [comment]
May I have this issue?

**Comment by SiddharthaAnand** [comment]
Hi,
I am a beginner.  Can I pick up this issue to contribute if noone else is looking into this ?

**Comment by aitoehigie** [comment]
Sent in a PR for this

**Comment by Jane222** [comment]
Hey, seeing as the last pull request hasn't been accepted for quite some time, mind if I submit my own? This would be my first one...

**Comment by dangra** [comment]
Fixed by #3011
